importer/src/main/java/org/hiero/mirror/importer/downloader/block/CompositeBlockSource.java

```
@@ -0,0 +1,103 @@
+
+    private final SourceHealth blockFileSourceHealth;
+    private final SourceHealth blockNodeSubscriberSourceHealth;
+    private final BlockStreamVerifier blockStreamVerifier;
+    private SourceHealth current;
+    private final BlockProperties properties;
+
+    public CompositeBlockSource(
+            BlockFileSource blockFileSource,
+            BlockNodeSubscriber blockNodeSubscriber,
+            BlockStreamVerifier blockStreamVerifier,
+            BlockProperties properties) {
+        this.blockFileSourceHealth = new SourceHealth(blockFileSource, BlockSourceType.FILE);
+        this.blockNodeSubscriberSourceHealth = new SourceHealth(blockNodeSubscriber, BlockSourceType.BLOCK_NODE);
+        this.blockStreamVerifier = blockStreamVerifier;
+        this.current = blockNodeSubscriberSourceHealth;
+        this.properties = properties;
+    }
+
+    @Override
+    @Leader
+    @Scheduled(fixedDelayString = "#{@blockProperties.getFrequency().toMillis()}")
+    public void get() {
+        if (!properties.isEnabled()) {
+            return;
+        }
+
+        var sourceHealth = getSourceHealth();
+        try {
+            sourceHealth.getSource().get();
+            sourceHealth.reset();
+        } catch (Throwable t) {
+            log.error("Failed to get block from {} source", sourceHealth.getType(), t);
+            sourceHealth.onError();
+        }
+    }
+
+    private SourceHealth getSourceHealth() {
+        return switch (properties.getSourceType()) {
+            case AUTO -> {
+                boolean switched = blockStreamVerifier
+                        .getLastBlockFilename()
+                        .map(BlockFile::isStreamedFilename)
+                        .orElse(false);
+                if (switched) {
+                    yield blockNodeSubscriberSourceHealth;
+                }
+
+                if (properties.getNodes().isEmpty()) {
+                    yield blockFileSourceHealth;
```

Maybe we should throw an error instead of assume they just want files?

---

reason is the default is `AUTO` and the logic is for when there's only block files in cloud storage.